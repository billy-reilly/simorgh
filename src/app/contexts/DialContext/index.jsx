import React from 'react';
import { any, objectOf, node } from 'prop-types';

export const DialContext = React.createContext({});

export const DialContextProvider = ({ dials, children }) => (
  <DialContext.Provider value={dials}>{children}</DialContext.Provider>
);

DialContextProvider.propTypes = {
  dials: objectOf(any).isRequired,
  children: node.isRequired,
};
