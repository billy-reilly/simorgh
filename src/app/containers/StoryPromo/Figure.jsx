import React, { useContext } from 'react';
import Image, { AmpImg } from '@bbc/psammead-image';
import ImagePlaceholder from '@bbc/psammead-image-placeholder';
import { storyItemImage } from '../../models/propTypes/storyItem';
import { RequestContext } from '../../contexts/RequestContext';

const StoryPromoFigure = ({
  path,
  altText,
  height,
  width,
  copyrightHolder,
}) => {
  const { platform } = useContext(RequestContext);
  if (!path || !altText || !width || !height) {
    return null;
  }

  const ratio = (height / width) * 100;
  const src = `https://ichef.bbci.co.uk/news/660${path}`;

  return (
    <ImagePlaceholder ratio={ratio}>
      {platform === 'amp' ? (
        <AmpImg
          alt={altText}
          attribution={copyrightHolder}
          layout="responsive"
          src={src}
          height={height}
          width={width}
        />
      ) : (
        <Image alt={altText} src={src} width={width} />
      )}
    </ImagePlaceholder>
  );
};

StoryPromoFigure.propTypes = storyItemImage;

export default StoryPromoFigure;
