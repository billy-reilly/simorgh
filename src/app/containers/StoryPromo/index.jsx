import React, { Fragment, useContext } from 'react';
import { shape } from 'prop-types';
import StoryPromoComponent, {
  Headline,
  Summary,
  Link,
} from '@bbc/psammead-story-promo';
import { storyItem } from '../../models/propTypes/storyItem';
import StoryPromoFigure from './Figure';
import Timestamp from '../Timestamp';
import { ServiceContext } from '../../contexts/ServiceContext';
import deepGet from '../../helpers/json/deepGet';

const StoryPromo = ({ item }) => {
  const { script } = useContext(ServiceContext);
  const headline = deepGet(['headlines', 'headline'], item);
  const url = deepGet(['locators', 'assetUri'], item);
  const summary = deepGet(['summary'], item);
  const timestamp = deepGet(['timestamp'], item);
  const imageValues = deepGet(['indexImage'], item);

  const Image = imageValues && <StoryPromoFigure {...imageValues} />;

  if (!headline || !url) {
    return null;
  }

  const Info = (
    <Fragment>
      {headline && (
        <Headline script={script}>
          <Link href={url}>{headline}</Link>
        </Headline>
      )}
      {summary && <Summary script={script}>{summary}</Summary>}
      {timestamp && (
        <Timestamp
          timestamp={timestamp * 1000}
          dateTimeFormat="YYYY-MM-DD"
          format="D MMMM YYYY"
        />
      )}
    </Fragment>
  );

  return <StoryPromoComponent image={Image} info={Info} />;
};

StoryPromo.propTypes = {
  item: shape(storyItem).isRequired,
};

export default StoryPromo;
