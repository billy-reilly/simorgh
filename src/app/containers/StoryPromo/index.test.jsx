import React from 'react';
import { render } from 'react-testing-library';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import { RequestContextProvider } from '../../contexts/RequestContext';
import { ServiceContextProvider } from '../../contexts/ServiceContext';
import deepClone from '../../helpers/json/deepClone';
import StoryPromo from '.';

let item;
const completeItem = {
  headlines: {
    headline: 'A headline',
  },
  locators: {
    assetUri: 'https://www.bbc.co.uk',
  },
  summary: 'Summary text',
  timestamp: 1556795033,
  indexImage: {
    path: '/cpsprodpb/0A06/production/image.jpg',
    height: 1152,
    width: 2048,
    altText: 'Image Alt text',
    copyrightHolder: 'Image provider',
  },
};

describe('StoryPromo Container', () => {
  shouldMatchSnapshot(
    'should render correctly for canonical',
    <ServiceContextProvider service="igbo">
      <RequestContextProvider
        platform="canonical"
        isUK
        origin="https://www.bbc.co.uk"
        id="c0000000000o"
        statsDestination="NEWS_PS_TEST"
        statsPageIdentifier="news.articles.c0000000000o"
      >
        <StoryPromo item={completeItem} />
      </RequestContextProvider>
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should render correctly for amp',
    <ServiceContextProvider service="igbo">
      <RequestContextProvider
        platform="amp"
        isUK
        origin="https://www.bbc.co.uk"
        id="c0000000000o"
        statsDestination="NEWS_PS_TEST"
        statsPageIdentifier="news.articles.c0000000000o"
      >
        <StoryPromo item={completeItem} />
      </RequestContextProvider>
    </ServiceContextProvider>,
  );

  describe('assertion tests', () => {
    beforeEach(() => {
      item = deepClone(completeItem);
    });

    it('should render h3, a, p, time', () => {
      const { container } = render(<StoryPromo item={item} />);

      expect(container.querySelectorAll('h3 a')[0].innerHTML).toEqual(
        item.headlines.headline,
      );
      expect(container.getElementsByTagName('p')[0].innerHTML).toEqual(
        item.summary,
      );
      expect(container.getElementsByTagName('time')[0].innerHTML).toEqual(
        '2 May 2019',
      );
    });

    it('should render img with src & alt when platform is canonical', () => {
      const { container } = render(
        <RequestContextProvider
          platform="canonical"
          isUK
          origin="https://www.bbc.co.uk"
          id="c0000000000o"
          statsDestination="NEWS_PS_TEST"
          statsPageIdentifier="news.articles.c0000000000o"
        >
          <StoryPromo item={item} />
        </RequestContextProvider>,
      );

      expect(container.getElementsByTagName('img').length).toEqual(1);
      expect(container.getElementsByTagName('amp-img').length).toEqual(0);
      expect(
        container.getElementsByTagName('img')[0].getAttribute('src'),
      ).toEqual(`https://ichef.bbci.co.uk/news/660${item.indexImage.path}`);
      expect(
        container.getElementsByTagName('img')[0].getAttribute('alt'),
      ).toEqual(item.indexImage.altText);
    });

    it('should render amp-img with src & alt when platform is amp', () => {
      const { container } = render(
        <RequestContextProvider
          platform="amp"
          isUK
          origin="https://www.bbc.co.uk"
          id="c0000000000o"
          statsDestination="NEWS_PS_TEST"
          statsPageIdentifier="news.articles.c0000000000o"
        >
          <StoryPromo item={item} />
        </RequestContextProvider>,
      );

      expect(container.getElementsByTagName('amp-img').length).toEqual(1);
      expect(container.getElementsByTagName('img').length).toEqual(0);
      expect(
        container.getElementsByTagName('amp-img')[0].getAttribute('src'),
      ).toEqual(`https://ichef.bbci.co.uk/news/660${item.indexImage.path}`);
      expect(
        container.getElementsByTagName('amp-img')[0].getAttribute('alt'),
      ).toEqual(item.indexImage.altText);
    });

    describe('With no headline provided', () => {
      beforeEach(() => {
        delete item.headlines;
      });

      it('should not include a headline element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('h3').length).toEqual(0);
      });
    });

    describe('With no summary provided', () => {
      beforeEach(() => {
        delete item.summary;
      });

      it('should not include a paragraph element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('p').length).toEqual(0);
      });
    });

    describe('With no timestamp provided', () => {
      beforeEach(() => {
        delete item.timestamp;
      });

      it('should not include a time element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('time').length).toEqual(0);
      });
    });

    describe('With no indexImage provided', () => {
      beforeEach(() => {
        delete item.indexImage;
      });

      it('should not include an img element', () => {
        const { container } = render(<StoryPromo item={item} />);

        expect(container.getElementsByTagName('img').length).toEqual(0);
      });
    });
  });
});
