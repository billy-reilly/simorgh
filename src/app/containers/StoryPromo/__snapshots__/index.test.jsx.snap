// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryPromo Container should render correctly for amp 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  max-width: 33.33%;
}

.c3 {
  display: inline-block;
  vertical-align: top;
  padding: 0 0.5rem;
  max-width: 66.67%;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c6 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c5 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c5:hover,
.c5:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5:visited {
  color: #6E6E73;
}

.c2 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #ECEAE7;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 30%;
  padding-bottom: 56.2500%;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

.c7 {
  font-size: 0.875rem;
  line-height: 1rem;
  color: #757575;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  padding-bottom: 0.25rem;
}

.c7:last-child {
  padding-bottom: 1rem;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    max-width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    max-width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c3 {
    max-width: 66.67%;
  }
}

@supports (display:grid) {
  .c3 {
    display: block;
    max-width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c6 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c6 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c6 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.8125rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <amp-img
        alt="Image Alt text"
        attribution="Image provider"
        height={1152}
        layout="responsive"
        src="https://ichef.bbci.co.uk/news/660/cpsprodpb/0A06/production/image.jpg"
        width={2048}
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <h3
      className="c4"
    >
      <a
        className="c5"
        href="https://www.bbc.co.uk"
      >
        A headline
      </a>
    </h3>
    <p
      className="c6"
    >
      Summary text
    </p>
    <time
      className="c7"
      dateTime="2019-05-02"
    >
      2 May 2019
    </time>
  </div>
</div>
`;

exports[`StoryPromo Container should render correctly for canonical 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: inline-block;
  vertical-align: top;
  max-width: 33.33%;
}

.c4 {
  display: inline-block;
  vertical-align: top;
  padding: 0 0.5rem;
  max-width: 66.67%;
}

.c5 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #222222;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
  font-weight: 700;
}

.c7 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #3F3F42;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  margin: 0;
  padding-bottom: 0.5rem;
}

.c6 {
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c6:hover,
.c6:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:visited {
  color: #6E6E73;
}

.c3 {
  display: block;
  width: 100%;
}

.c2 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #ECEAE7;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: 30%;
  padding-bottom: 56.2500%;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

.c8 {
  font-size: 0.875rem;
  line-height: 1rem;
  color: #757575;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  padding-bottom: 0.25rem;
}

.c8:last-child {
  padding-bottom: 1rem;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    grid-template-columns: repeat(6,1fr);
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    max-width: 33.33%;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
    max-width: initial;
    grid-column: 1 / span 2;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding: 0 1rem;
  }
}

@media (min-width:80rem) {
  .c4 {
    max-width: 66.67%;
  }
}

@supports (display:grid) {
  .c4 {
    display: block;
    max-width: initial;
    padding: initial;
    grid-column: 3 / span 4;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c5 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c7 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c7 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (max-width:37.4375rem) {
  .c7 {
    display: none;
    visibility: hidden;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c8 {
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c8 {
    font-size: 0.8125rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="Image Alt text"
        className="c3"
        height={null}
        src="https://ichef.bbci.co.uk/news/660/cpsprodpb/0A06/production/image.jpg"
        width={2048}
      />
    </div>
  </div>
  <div
    className="c4"
  >
    <h3
      className="c5"
    >
      <a
        className="c6"
        href="https://www.bbc.co.uk"
      >
        A headline
      </a>
    </h3>
    <p
      className="c7"
    >
      Summary text
    </p>
    <time
      className="c8"
      dateTime="2019-05-02"
    >
      2 May 2019
    </time>
  </div>
</div>
`;
